File: examples/rogue/character.js
// Rogue — fragile, fast, gets in and gets out.
//
// Build budget walkthrough (~95 / 100):
//   stats: speed 300 (30), small size 22 (~7), 2 extra slots (8) → 45
//   Backstab ~31   Blink ~6   Knife Fan ~15
//
// Tradeoffs:
//   + huge mobility, hard to hit, invulnerable dash
//   - low HP, single weakness moment between dashes

function buildCharacter() {
  const backstab = {
    name: "Backstab",
    key: "space",
    cooldownMs: 800,
    cast: {
      kind: "melee", color: "purple",
      range: 50, arcDeg: 60,
      onHit: [
        { effect: "damage", amount: 22 },
        { effect: "stun", durationMs: 250 },
      ],
    },
  };

  const blink = {
    name: "Blink",
    key: "f",
    cooldownMs: 3500,
    cast: {
      kind: "dash", color: "white",
      distance: 220, durationMs: 200, invulnerable: true,
    },
  };

  const knives = {
    name: "Knife Fan",
    key: "q",
    cooldownMs: 1800,
    cast: {
      kind: "projectile", color: "silver",
      speed: 560, radius: 4, lifetimeMs: 900,
      count: 3, spreadDeg: 25,
      onHit: [{ effect: "damage", amount: 10 }],
    },
  };

  return {
    characterName: "Rogue",
    color: "purple",
    size: 22,
    speed: 300,
    maxHealth: 70,
    powers: [backstab, blink, knives],
  };
}

File: examples/rogue/character.py
def build_character():
    # Rogue — high-mobility skirmisher.
    # Demonstrates: DASH + MELEE backstab + thrown PROJECTILE.

    blink = {
        "name": "Blink",
        "key": "shift",
        "cooldownMs": 2200,
        "cast": {
            "kind": "dash",
            "color": "#a070ff",
            "distance": 240,
            "durationMs": 180,
            "invulnerable": True,
        },
    }

    backstab = {
        "name": "Backstab",
        "key": "space",
        "cooldownMs": 600,
        "cast": {
            "kind": "melee",
            "color": "yellow",
            "range": 40,
            "arcDeg": 70,
            "onHit": [
                {"effect": "damage", "amount": 28},
                {"effect": "stun", "durationMs": 250},
            ],
        },
    }

    throwing_knife = {
        "name": "Throwing Knife",
        "key": "e",
        "cooldownMs": 500,
        "cast": {
            "kind": "projectile",
            "color": "silver",
            "speed": 700,
            "radius": 4,
            "lifetimeMs": 800,
            "count": 1,
            "spreadDeg": 0,
            "pierce": True,
            "onHit": [{"effect": "damage", "amount": 12}],
        },
    }

    return {
        "characterName": "Rogue",
        "color": "#a070ff",
        "size": 20,
        "speed": 320,
        "maxHealth": 80,
        "powers": [backstab, blink, throwing_knife],
    }
